koala/ast/graph.py

```diff
@@ -218,15 +221,31 @@ def children(self,ast):
     def parent(self,ast):
         args = ast.successors(self)
         return args[0] if args else None
-    
+
+    def find_special_function(self, ast):
+        found = False
+        current = self
+
+        while current is not None:
+            # print 'VERIF', current.tvalue.lower()
+
+            if current.tvalue.lower() == 'sumproduct':
+                found = True
+                break
+            else:
+                current = current.parent(ast)
+
+        return found
```

Comment:
In my view, that's the best solution to find now a previous call of sumproduct.
But I think we must keep an eye on the execution times, it could be greedy.